test/e2e/providers/vsphere/vsphere.go

```diff
@@ -133,9 +140,271 @@ func (p *Provider) GetType() config.PlatformType {
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
+			},
+			StorageClassName: &sc.Name,
+		},
+	}
+	return client.CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), &pvcSpec, meta.CreateOptions{})
+}
+
+// ensureStorageClass ensures a usable vSphere NTFS storage class exists
+func (p *Provider) ensureStorageClass(client client.Interface) (*storage.StorageClass, error) {
+	sc, err := client.StorageV1().StorageClasses().Get(context.TODO(), storageClassName, meta.GetOptions{})
+	if err == nil {
+		return sc, nil
+	} else if !k8sapierrors.IsNotFound(err) {
+		return nil, fmt.Errorf("error getting storage class '%s': %w", storageClassName, err)
+	}
+	volumeBinding := storage.VolumeBindingImmediate
+	reclaimPolicy := core.PersistentVolumeReclaimDelete
+	sc = &storage.StorageClass{
+		ObjectMeta: meta.ObjectMeta{
+			Name: storageClassName,
+		},
+		Provisioner:       "csi.vsphere.vmware.com",
+		Parameters:        map[string]string{"fstype": "ntfs"},
+		ReclaimPolicy:     &reclaimPolicy,
+		VolumeBindingMode: &volumeBinding,
+	}
+	return client.StorageV1().StorageClasses().Create(context.TODO(), sc, meta.CreateOptions{})
 }
 
-func (p *Provider) CreatePVC(_ client.Interface, _ string) (*core.PersistentVolumeClaim, error) {
-	return nil, fmt.Errorf("storage not supported on vSphere")
+// ensureWindowsCSIDrivers ensures that the vSphere CSI drivers are deployed across Windows nodes
+func (p *Provider) ensureWindowsCSIDrivers(client client.Interface) error {
+	csiNamespace := "openshift-cluster-csi-drivers"
+	// The FSS used by Linux nodes is unusable as it does not set csi-windows-support to true.
+	windowsFSS := "win-internal-feature-states.csi.vsphere.vmware.com"
+	fssCM := core.ConfigMap{
+		ObjectMeta: meta.ObjectMeta{
+			Name: windowsFSS,
+		},
+		Data: map[string]string{
+			"block-volume-snapshot":            "true",
+			"cnsmgr-suspend-create-volume":     "true",
+			"csi-auth-check":                   "true",
+			"csi-migration":                    "true",
+			"improved-csi-idempotency":         "true",
+			"improved-volume-topology":         "false",
+			"online-volume-extend":             "true",
+			"topology-preferential-datastores": "true",
+			"csi-windows-support":              "true",
+		},
+		BinaryData: nil,
+	}
+	_, err := client.CoreV1().ConfigMaps(csiNamespace).Create(context.TODO(), &fssCM, meta.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not create FSS ConfigMap: %w", err)
+	}
+
+	dsName := "vmware-vsphere-csi-driver-node-windows"
+	directoryType := core.HostPathDirectory
+	directoryOrCreate := core.HostPathDirectoryOrCreate
+	ds := apps.DaemonSet{
+		ObjectMeta: meta.ObjectMeta{
+			Name: dsName,
+		},
+		Spec: apps.DaemonSetSpec{
+			Selector: &meta.LabelSelector{MatchLabels: map[string]string{"app": dsName}},
+			Template: core.PodTemplateSpec{
+				ObjectMeta: meta.ObjectMeta{
+					Labels: map[string]string{"app": dsName},
+				},
+				Spec: core.PodSpec{
```

Comment:
Please add the `os` field. In addition lets ensure that no warnings are thrown because of this DaemonSet. I really would like the rest of the workloads cleaned up but don't want to add to that list.

Reply:
Do you see warnings like?
```
W0503 04:31:12.362569   15534 warnings.go:70] would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "hybrid-overlay-query2" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "hybrid-overlay-query2" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "hybrid-overlay-query2" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Reply:
Still working on checking this

Reply:
Now the only warning is coming from the webserver creation. I'll handle that in a separate PR.